webpack/scenes/ContentViews/Details/Versions/ContentViewVersions.js

```diff
@@ -137,7 +182,19 @@ const ContentViewVersions = ({ cvId }) => {
       autocompleteEndpoint={`/content_view_versions/auto_complete_search?content_view_id=${cvId}`}
       fetchItems={useCallback(params => getContentViewVersions(cvId, params), [cvId])}
       additionalListeners={[pollingFinished]}
-    />);
+    >
+      {promoting &&
```

Comment:
this seems redundant: perhaps instead of conditionally showing it here, you could just do a check on the ContentViewVersion like: 
if (!show) return null;
at the top of the functional scope. 

You could then remove the conditional {promoting && <thing/>} logic.

Reply:
The conditional return null at top of functional scope makes react hook rules unhappy. It would have to go below state, useEffects and other hooks in the file. Given that, I think we should stick to the pattern of conditional rendering..Thoughts?

`React Hook "useState" is called conditionally. React Hooks must be called in the exact same order in every component render`

Reply:
As discussed, we'll likely be changing our modal pattern to: 
https://github.com/Katello/katello/pull/9553/files#diff-dc6c9c6fb2119021430864e20e82157272fb9a2a55accf48285743eb8781353fR66
https://github.com/Katello/katello/pull/9553/files#diff-04ef9c1add8b6ffd04659bb6b4e9c32bc4bea75f751dd4d392573ebe09981d2dR97-R101